src/app/calendar/_components/timetable/TimeTableWeekdayColumn.tsx

```diff
@@ -0,0 +1,81 @@
+"use client";
+import { twMerge } from "tailwind-merge";
+import { ReservationWithTimestamp } from "./TimeTableMainPart";
+import { Reservation } from "../../page";
+import { useEffect, useState } from "react";
+
+type TimeTableDayContentProps = {
+  reservationTimeStamps: ReservationWithTimestamp[];
+  day: string;
+  typeFilters: string[];
+  selectedRoom: string;
+};
+
+export function TimeTableDayContent({
+  reservationTimeStamps,
+  day,
+  typeFilters,
+  selectedRoom,
+}: TimeTableDayContentProps) {
+  const roomFilters = ["2.41"];
+
+  const filteredReservations = filterReservationsWithProperTypeAndRoom(
+    reservationTimeStamps,
+    typeFilters,
+    selectedRoom,
+  );
+
+  function filterReservationsWithProperTypeAndRoom(
+    reservationTimeStamps: ReservationWithTimestamp[],
+    typeFilters: string[],
+    roomFilter: string,
+  ) {
+    return reservationTimeStamps.map(
+      (reservation: ReservationWithTimestamp) => {
+        if (!reservation.reservation) {
+          return {
+            reservation: null,
+            timestamp: reservation.timestamp,
+          };
+        }
+        return roomFilter === reservation.reservation?.classroom.name &&
+          typeFilters.includes(reservation.reservation?.type)
+          ? {
+              reservation: reservation.reservation,
+              timestamp: reservation.timestamp,
+            }
+          : {
+              reservation: null,
+              timestamp: reservation.timestamp,
+            };
+      },
+    );
+  }
+  return (
+    <div className="pt-2">
+      {filteredReservations.map((reservation, j) => (
+        <div
+          key={j + day}
```

Comment:
it's generally not a good idea to use index inside React keys, if you have some reservation id or unique name it's better to use it 

https://dev.to/shiv1998/why-not-to-use-index-as-key-in-react-lists-practical-example-3e66